ui/ui-config/pages/app/new/index.tsx

```diff
@@ -20,87 +24,96 @@ class CustomErrorHandler extends EventDispatcher<{ message: string }> {
       this.message = message;
 
       if (this.message?.config?.url === "/api/v1/status/public") {
-        console.info(`could not reach API: ${message}`);
+        console.info(`Could not reach API: ${message}`);
       } else {
         errorHandler.set({ message: this.message });
       }
     });
   }
 }
 
-export default function Page(props) {
-  const {
-    plugins,
-  }: {
-    plugins: Actions.PluginsList["plugins"];
-  } = props;
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
 
+export const getServerSideProps = async (ctx: GetServerSidePropsContext) => {
+  const client = generateClient(ctx);
+  const { plugins } = await client.request<Actions.PluginsList>(
+    "get",
+    `/plugins`,
+    {
+      includeInstalled: true,
+      includeAvailable: true,
+      includeVersions: false,
+    }
+  );
+  return { props: { plugins } };
+};
+
+const Page: NextPageWithInferredProps<typeof getServerSideProps> = ({
+  plugins,
+}) => {
   const router = useRouter();
   const { client } = useApi();
-  const [plugin, setPlugin] = useState<
-    Partial<Actions.PluginsList["plugins"][number]>
-  >({ name: "" });
-  const [cards, setCards] = useState([]);
+  const [plugin, setPlugin] = useState<Partial<PluginWithVersion>>(() => ({
+    name: "",
+  }));
+  const [cards, setCards] = useState<PluginWithVersion[]>([]);
   const [loading, setLoading] = useState(false);
   const [showSpinner, setShowSpinner] = useState(true);
-  const [installingMessage, setInstallingMessage]: [
-    string | boolean,
-    Function
-  ] = useState(false);
+  const [installingMessage, setInstallingMessage] = useState<React.ReactNode>();
 
   async function resetPluginsAndApps() {
     const { plugins: _plugins }: Actions.PluginsList = await client.request(
       "get",
       `/plugins`
     );
-    prepareCards(_plugins);
+    return prepareCards(_plugins);
   }
 
   function prepareCards(_plugins: Actions.PluginsList["plugins"]) {
-    setCards(
-      _plugins
-        .filter((p) => p.source || p.destination)
-        .sort((a, b) => (a.name > b.name ? 1 : -1))
-        .sort((a, b) => {
-          if (a.installed === b.installed) return 0;
-          if (a.installed) return -1;
-          return 1;
-        })
-    );
+    const nextCards = _plugins
+      .filter((p) => p.source || p.destination)
+      .sort((a, b) => (a.name > b.name ? 1 : -1))
```

Comment:
```suggestion
      .sort((a, b) => (a.name - b.name))
```
I _think_ this will function more consistently.